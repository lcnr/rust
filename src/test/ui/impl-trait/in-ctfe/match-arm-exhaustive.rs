// This previously compiled, but broke with #101478.
//
// See that PR for more details.
trait MyTrait: Copy {
    const ASSOC: u8;
}

impl MyTrait for () {
    const ASSOC: u8 = 0;
}

const fn yeet() -> impl MyTrait {}

const fn output<T: MyTrait>(_: T) -> u8 {
    <T as MyTrait>::ASSOC
}

const CT: u8 = output(yeet());

fn main() {
    match 0 {
        CT => (),
        1.. => (),
    }
}
