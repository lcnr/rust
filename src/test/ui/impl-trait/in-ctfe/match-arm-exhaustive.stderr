error: constant pattern depends on a generic parameter
  --> $DIR/match-arm-exhaustive.rs:19:9
   |
LL |         CT => (),
   |         ^^

error: constant pattern depends on a generic parameter
  --> $DIR/match-arm-exhaustive.rs:19:9
   |
LL |         CT => (),
   |         ^^

error: aborting due to 2 previous errors

